### PR TITLE
Fix #1568: NextStep OpenID Connect (OIDC) Support

### LINKED ISSUE #1568: NextStep OpenID Connect (OIDC) Support
Support authentication via OpenID Connect (OIDC). Because of the backward compatibility, there should be no authentication by default.

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

**Review by romanstrobl — APPROVED** [review]
Looks OK.

File: powerauth-nextstep/src/main/resources/application.properties [hunk 1]
```diff
@@ -70,5 +70,15 @@ banner.application.version=@project.version@
 
 logging.config=${POWERAUTH_NEXTSTEP_LOGGING:}
 
+# OpenID Connect (OIDC) Settings
+#powerauth.nextstep.security.auth.type=OIDC
+#spring.security.oauth2.client.registration.nextstep-oidc-client.provider=nextstep-oidc-provider
```
Comment (banterCZ) [inline powerauth-nextstep/src/main/resources/application.properties:75]: Autoconfiguration has to be improved not to fail when the property is specified as empty and the type is not OIDC. In that case, the application should normally start.

File: powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/configuration/SecurityConfig.java [hunk 2]
```diff
@@ -31,19 +39,57 @@
...
      * @param http HTTP security.
      * @throws Exception Thrown when configuration fails.
      */
     @Bean
     public SecurityFilterChain filterChain(final HttpSecurity http) throws Exception {
+        if (authType == AuthType.OIDC) {
+            logger.info("Initializing OIDC authentication.");
+            http.authorizeHttpRequests(authorize -> authorize
+                            .requestMatchers(
+                                    new AntPathRequestMatcher("/login/oauth2/**"),
```
Comment (zcgandcomp) [inline powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/configuration/SecurityConfig.java:60]: Only 

- actuator
- status 
- and the  /login/ouath2
should be permitted.


Swagger console and docs should be accessible only for authorized user (same as in PAC)

**Review by zcgandcomp — COMMENTED** [review]
We should enable only minimum endpoint without authorization.

**Review by zcgandcomp — APPROVED** [review]
Looks OK.

**Review by romanstrobl — APPROVED** [review]

**Pull request merged**